NFIQ2/NFIQ2Api/nfiq2api.cpp

```diff
@@ -93,7 +94,8 @@ InitNfiq2()
 			    new NFIQ::NFIQ2Algorithm(GetYamlFilePath(),
 				"ccd75820b48c19f1645ef5e9c481c592"));
 #endif
-			return g_nfiq2->getParameterHash().c_str();
+			strncpy(paramHash, g_nfiq2->getParameterHash().c_str(), 33);
```

Comment:
This works, but since you have other changes required, it popped into my head that we can be safer and easier. Use `strndup` instead. User doesn't have to allocate the pointer ahead of time, we do. 

User code:
```
char *hash = NULL;
if (InitNfiq2(&hash) != NULL)
    printf("%s\n", hash);
```

In the case of unforeseen error, we use `getParameterHash().length() + 1` for the size parameter instead of `33`.

Reply:
Turns out strndup is POSIX so it doesn't exist on windows (why appveyor is failing). We might have to keep it so that the user needs to allocate the buffer themselves.